### PR TITLE
frontend/feat/#1063: added clevertap events for cust and driver

### PR DESCRIPTION
**Pull request by ashkriti**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [x] Enhancement
- [x] Refactoring
- [ ] Dependency updates

## Description
Added the Clevertap events for the customer and driver

## Motivation and Context
https://github.com/nammayatri/nammayatri/issues/1063

## How did you test it?
Tested Manually

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Frontend/ui-common/src/Engineering/Helpers/LogEvent.purs [hunk 1]
```diff
@@ -81,3 +89,21 @@ logEventWithTwoParams logField event key1 value1 key2 value2  = do
   logDestinations <- getLogDestination
   void $ foreachE logDestinations (triggerLogWithTwoParams event logField key1 value1 key2 value2)
   pure unit 
+
+triggerLogEventWithMultipleParams :: String -> Object Foreign -> Array ClevertapEventParams -> String -> Effect Unit
+triggerLogEventWithMultipleParams event logField arr logDestination = case logDestination of
+  "CLEVERTAP" -> do
+    _ <- cleverTapEvent event arr
+    pure unit
```
Comment (ashishsingh101) [inline Frontend/ui-common/src/Engineering/Helpers/LogEvent.purs:97]: use `void` and remove pure unit

File: Frontend/ui-common/src/Engineering/Helpers/LogEvent.purs [hunk 2]
```diff
@@ -81,3 +89,21 @@ logEventWithTwoParams logField event key1 value1 key2 value2  = do
...
+  _ -> pure unit
+
+logEventWithMultipleParams :: Object Foreign -> String -> Array ClevertapEventParams -> Effect Unit
+logEventWithMultipleParams logField event arr = do
+  logDestinations <- getLogDestination
+  void $ foreachE logDestinations (triggerLogEventWithMultipleParams event logField arr)
+  pure unit
```
Comment (ashishsingh101) [inline Frontend/ui-common/src/Engineering/Helpers/LogEvent.purs:104]: no need for `pure unit`

**Review by ashishsingh101 — CHANGES_REQUESTED** [review]

File: Frontend/ui-common/src/Engineering/Helpers/LogEvent.purs [hunk 3]
```diff
@@ -81,3 +89,21 @@ logEventWithTwoParams logField event key1 value1 key2 value2  = do
...
+
+type ClevertapEventParams = {
+  key :: String ,
+  value :: String
+}
```
Comment (ashishsingh101) [inline Frontend/ui-common/src/Engineering/Helpers/LogEvent.purs:109]: don't define same `type` multiple times

File: Frontend/ui-customer/src/Flow.purs [hunk 4]
```diff
@@ -974,10 +989,21 @@ homeScreenFlow = do
                                         let (RideBookingAPIDetails bookingDetails) = resp.bookingDetails
                                         let (RideBookingDetails contents) = bookingDetails.contents
                                         let (RideAPIEntity ride) = fromMaybe dummyRideAPIEntity (resp.rideList !! 0)
+                                        let (FareBreakupAPIEntity fareBreakup) = fromMaybe HomeScreenData.dummyFareBreakUp (resp.fareBreakup !! 0)
```
Comment (ashishsingh101) [inline Frontend/ui-customer/src/Flow.purs:992]: remove if not using

**Review by ashishsingh101 — CHANGES_REQUESTED** [review]

File: Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/NotificationUtils.java [hunk 5]
```diff
@@ -384,8 +390,10 @@ public static void showNotification(Context context, String title, String msg, J
             Bundle params = new Bundle();
             mFirebaseAnalytics = FirebaseAnalytics.getInstance(context);
             if (key.equals("USER"))
-            {if (CANCELLED_PRODUCT.equals(notificationType))
-                mFirebaseAnalytics.logEvent("ny_user_ride_cancelled",params);
+            {if (CANCELLED_PRODUCT.equals(notificationType)) {
```
[the remark this answers is not in the document]
Comment (ashkriti) [reply Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/NotificationUtils.java:391]: Done

File: Frontend/ui-common/src/Engineering/Helpers/JBridge.js [hunk 6]
```diff
@@ -1734,6 +1736,16 @@ export const waitingCountdownTimer = function (startingTime) {
   };
 };
 
+export const cleverTapEvent = function(event){
```
[the remark this answers is not in the document]
Comment (ashkriti) [reply Frontend/ui-common/src/Engineering/Helpers/JBridge.js:1736]: Done

File: Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/NotificationUtils.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (Vignesh-772) [reply Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/NotificationUtils.java:391]: single function I meant will push the firebase log also. like we are doing in purescript.

**Review by Vignesh-772 — CHANGES_REQUESTED** [review]

**Review by Vignesh-772 — APPROVED** [review]

**Review by kavya-shree-s — APPROVED** [review]

**Review by ashishsingh101 — APPROVED** [review]

**Pull request merged**